src/wallet/wallet.go

```diff
@@ -102,6 +129,114 @@ type Options struct {
+	HoursSelectionTypeAuto = "auto"
+
+	// HoursSelectionModeShare will distribute coin hours equally amongst destinations
+	HoursSelectionModeShare = "share"
+)
+
+// HoursSelection defines options for hours distribution
+type HoursSelection struct {
+	Type        string
+	Mode        string
+	ShareFactor *decimal.Decimal
+}
+
+// CreateTransactionWalletParams defines a wallet to spend from and optionally which addresses in the wallet
+type CreateTransactionWalletParams struct {
+	ID        string
+	Addresses []cipher.Address
+	Password  []byte
+}
+
+// CreateTransactionParams defines control parameters for transaction construction
+type CreateTransactionParams struct {
+	HoursSelection HoursSelection
+	Wallet         CreateTransactionWalletParams
+	ChangeAddress  cipher.Address
+	To             []coin.TransactionOutput
+}
+
+// Validate validates CreateTransactionParams
+func (c CreateTransactionParams) Validate() error {
+	if c.ChangeAddress.Empty() {
+		return NewError(errors.New("ChangeAddress is required"))
+	}
+
+	if len(c.To) == 0 {
+		return NewError(errors.New("To is required"))
+	}
+
+	for _, to := range c.To {
+		if to.Coins == 0 {
+			return NewError(errors.New("To.Coins must not be zero"))
+		}
+
+		if to.Address.Empty() {
+			return NewError(errors.New("To.Address must not be empty"))
+		}
+	}
+
+	if c.Wallet.ID == "" {
+		return NewError(errors.New("Wallet.ID is required"))
+	}
+
+	for _, a := range c.Wallet.Addresses {
+		if a.Empty() {
+			return NewError(errors.New("Wallet.Addresses must not contain an empty value"))
+		}
+	}
+
+	switch c.HoursSelection.Type {
+	case HoursSelectionTypeAuto:
+		for _, to := range c.To {
+			if to.Hours != 0 {
+				return NewError(errors.New("To.Hours must be zero for auto type hours selection"))
+			}
+		}
+
+		if c.HoursSelection.Mode == "" {
+			return NewError(errors.New("HoursSelection.Mode is required for auto type hours selection"))
+		}
+
+		switch c.HoursSelection.Mode {
+		case HoursSelectionModeShare:
+		default:
+			return NewError(errors.New("Invalid HoursSelection.Mode"))
+		}
+
+	case HoursSelectionTypeManual:
+		if c.HoursSelection.Mode != "" {
+			return NewError(errors.New("HoursSelection.Mode cannot be used for manual type hours selection"))
+		}
+
+	default:
+		return NewError(errors.New("Invalid HoursSelection.Type"))
+	}
+
+	if c.HoursSelection.ShareFactor == nil {
+		if c.HoursSelection.Mode == HoursSelectionModeShare {
+			return NewError(errors.New("HoursSelection.ShareFactor must be set for share mode"))
+		}
+	} else {
+		if c.HoursSelection.Mode != HoursSelectionModeShare {
+			return NewError(errors.New("HoursSelection.ShareFactor can only be used for share mode"))
+		}
+
+		zero := decimal.New(0, 0)
+		one := decimal.New(1, 0)
+		if c.HoursSelection.ShareFactor.LessThan(zero) || c.HoursSelection.ShareFactor.GreaterThan(one) {
+			return NewError(errors.New("HoursSelection.ShareFactor must be >= 0 and <= 1"))
+		}
```

Comment:
I think its better to be explicit about the error like in the other `Validate` method
```
switch {
  case r.HoursSelection.ShareFactor.LessThan(decimal.New(0, 0)):
      return errors.New("hours_selection.share_factor cannot be negative")
  case r.HoursSelection.ShareFactor.GreaterThan(decimal.New(1, 0)):
      return errors.New("hours_selection.share_factor cannot be more than 1")
}
```